library/platform_util.c

```diff
@@ -65,3 +72,54 @@ void mbedtls_platform_zeroize( void *buf, size_t len )
     memset_func( buf, 0, len );
 }
 #endif /* MBEDTLS_PLATFORM_ZEROIZE_ALT */
+
+#if defined(MBEDTLS_HAVE_TIME_DATE) && !defined(MBEDTLS_PLATFORM_GMTIME_ALT)
+#include <time.h>
+#if !defined(_WIN32) && (defined(unix) || defined(__unix) || \
+    defined(__unix__) || (defined(__APPLE__) && defined(__MACH__)))
+#include <unistd.h>
+#if !defined(_POSIX_VERSION) || _POSIX_C_SOURCE > _POSIX_THREAD_SAFE_FUNCTIONS
+/*
+ * This is a convenience shorthand macro to avoid checking the long
+ * preprocessor conditions above. Ideally, we could expose this macro in
+ * platform_utils.h and simply use it in platform_utils.c, threading.c and
+ * threading.h. However, this macro is not part of the Mbed TLS public API, so
+ * we keep it private by only definining it in this file
+ */
+#define PLATFORM_UTIL_USE_GMTIME
+#endif /* !_POSIX_VERSION || _POSIX_C_SOURCE > _POSIX_THREAD_SAFE_FUNCTIONS */
+#endif /* !_WIN32 && (unix || __unix || __unix__ || (__APPLE__ && __MACH__)) */
+
+struct tm *mbedtls_platform_gmtime( const mbedtls_time_t *tt,
+                                    struct tm *tm_buf )
+{
+#if defined(_WIN32) && !defined(EFIX64) && !defined(EFI32)
+    return( ( gmtime_s( tm_buf, tt ) == 0 ) ? tm_buf : NULL );
+#elif defined(__IAR_SYSTEMS_ICC__)
+    return( gmtime_s( tt, tm_buf ) );
```

Comment:
I do not have access to this toolchain, so I dont really know if this works...

Reply:
I bet you need ```#define __STDC_WANT_LIB_EXT1__ 1``` somewhere before first ```#include <time.h>```, as the  protype of gmtime_s() is inside a ```#if __STDC_WANT_LIB_EXT1__ == 1``` block in IAR's time32.h.

Reply:
Oh I see, you are using the C11 Annex K extensions

Reply:
@TeroJaasko Using this C11 extension leads to problems, for example we get a compilation failure because `memset()` is being deprecated in favor of `memset_s()`. Also, if the extension is not available, we'd fall back to the default `gmtime()` implementation, which for Mbed OS is not thread-safe because `MBEDTLS_THREADING_C` is not defined. @TeroJaasko Is it acceptable to fall back to `gmtime()` non-Windows, non-POSIX, and thereby using a non-thread safe version of `gmtime()` in IAR by default, yet allowing to replace it by a custom, thread-safe version?